movement/roi/conditions.py

```diff
@@ -0,0 +1,105 @@
+"""Functions for computing condition arrays involving RoIs."""
+
+from collections import defaultdict
+from collections.abc import Sequence
+
+import numpy as np
+import xarray as xr
+
+from movement.roi.base import BaseRegionOfInterest
+
+
+def compute_region_occupancy(
+    data,
+    regions: Sequence[BaseRegionOfInterest],
+) -> xr.DataArray:
+    """Return a condition array indicating if points were inside regions.
+
+    The function returns a boolean DataArray where each element indicates
+    whether a point in the input ``data`` lies within the corresponding RoIs
+    in ``regions``. The original dimensions of ``data`` are preserved, except
+    for the ``space`` dimension which is replaced by the ``occupancy``
+    dimension. The ``occupancy`` dimension has a number of elements equal to
+    the number of RoIs in the ``regions`` argument and it's coordinate names
+    correspond to the names of the given RoIs.
```

Comment:
To me it would make more sense to name the new dimensions `region` instead of `occupancy`.

This makes sense to me because the coordinates of that new dim are region names, so doing `occupancies.sel(region="triangle")` is more natural and readable thatn `occupancies.sel(occupancy="triangle")`.

`occupancy` is the type of data being represented in this array, so it would more appropriate as the "name" of the variabe, i.e. maybe we should set it as the `.name` attribute of the returned array.